test/tools/yulInterpreter/EwasmBuiltinInterpreter.cpp

```diff
@@ -139,6 +167,11 @@ u256 EwasmBuiltinInterpreter::evalBuiltin(YulString _fun, vector<u256> const& _a
 		return clz64(arg[0] & uint32_t(-1)) - 32;
 	else if (_fun == "i64.clz"_yulstring)
 		return clz64(arg[0]);
+	else if (_fun == "i32.ctz"_yulstring)
+		// NOTE: the clz implementation assumes 64-bit inputs, hence the adjustment
+		return ctz64(arg[0] & uint32_t(-1)) - 32;
```

Comment:
Wait a second... this weird thing definitely works for leading zeroes... but for trailing zeroes :-)?
E.g. if there's no trailing zero this will return ``-32``, won't it?
So it would actually need to be even worse here... like:
```suggestion
		return ctz64(uint64_t(arg[0] & uint32_t(-1)) << 32) - 32;
```
...but that's not only bad, that's horrible :-). Maybe we should really just have ``ctz32``?

Reply:
I shouldn't really deal with PRs today...

Isn't `return ctz64(arg[0] & uint32_t(-1));` doing the right thing though?